pkg/util/label/label.go

```diff
@@ -0,0 +1,44 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package label
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// The AdditionalLabels is a big string consisting of key=value, set at build time
+// when set are to be added to Deployments, CronJob and KNativeService whose pods
+// should expose these labels
+
+// AdditionalLabels are labels=values, they MUST be set as key=value separated by comma ,
+// example: myKey1=myValue1,myKey2=myValue2
+// Also it supports replacing a value for the integration name, just put @ around it
+// example: myKey1=myValue1,myKey2=@integration@
+var AdditionalLabels = ""
+
+// parses the AdditionalLabels variable and returns as map[string]string
+func AddLabels(integration string) map[string]string {
+	AdditionalLabels = strings.ReplaceAll(AdditionalLabels, "@integration@", integration)
+	AdditionalLabels = fmt.Sprintf("%s=%s,%s", v1.IntegrationLabel, integration, AdditionalLabels)
```

Comment:
Tests fail because there is a trailing comma in case the additional labels are empty.

Reply:
There is a `TestParseEmptyAdditionalLabels` test which passes ok. At least looking at the failing tests, it seems to be unrelated to this change.

Reply:
That's because that is a unit test. Hereafter is an occurrence of the error in real tests:

```
`selector` does not match template `labels`","errorVerbose":"Deployment.apps \"rest-producer\" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{\"camel.apache.org/integration\":\"rest-consumer\"}: `selector` does not match template `labels`\nerror during apply resource: &Deployment{ObjectMeta:{rest-producer  test-0c75ba0a-aa50-4f8a-9876-b904cb172c1c    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[camel.apache.org/generation:1 camel.apache.org/integration:rest-producer] map[] [{camel.apache.org/v1 Integration rest-producer b309832d-acd5-467f-ba3c-599cae030559 0xc001b948c1 0xc001b948c2}] []  []},Spec:DeploymentSpec{Replicas:*1,Selector:&v1.LabelSelector{MatchLabels:map[string]string{camel.apache.org/integration: rest-producer,},MatchExpressions:[]LabelSelectorRequirement{},}
```